stratum/hal/lib/common/yang_parse_tree_paths.cc

```diff
@@ -460,6 +484,88 @@ TreeNodeEventHandler GetOnPollFunctor(
+    // Create a data retrieval request.
+    DataRequest req;
+    auto* request = req.add_requests()->mutable_optical_channel_info();
+    request->set_module(module);
+    request->set_network_interface(network_interface);
+    // In-place definition of method retrieving data from generic response
+    // and saving into 'resp' local variable.
+    OpticalChannelInfo resp{};
+    // Writer for retrieving value
+    DataResponseWriter writer([&resp](const DataResponse& in) {
+        if (!in.has_optical_channel_info()) return false;
+        resp = in.optical_channel_info();
+        return true;
+    });
+    // Query the switch. The returned status is ignored as there is no way to
+    // notify the controller that something went wrong. The error is logged when
+    // it is created.
+    // Here we ignore the node_id since it is not valid in this case.
+    tree->GetSwitchInterface()
+        ->RetrieveValue(/*node_id*/ 0, req, &writer, /* details= */ nullptr)
+        .IgnoreError();
+    // Return the retrieved value.
+    T value = (resp.*inner_message_get_field_func)();
+    return SendResponse(GetResponse(path, (*process_func)(value)), stream);
+  };
+}
+
+// Optical Port-specific version. Extra parameters needed:
+// - module index ('module')
+// - network interface index ('network_interface')
+template<typename T, typename U, typename V>
+TreeNodeEventHandler GetOpticalOnPollFunctor(
+    int32 module, int32 network_interface, YangParseTree* tree,
+    bool (OpticalChannelInfo::*has_inner_msg_func)() const,
+    const T& (OpticalChannelInfo::*get_inner_msg_func)() const,
+    U (T::*get_inner_field_func)() const,
+    V (*process_field_func)(const U&)) {
+  return [=](const GnmiEvent& event, const ::gnmi::Path& path,
+             GnmiSubscribeStream* stream) {
+    // Create a data retrieval request.
+    DataRequest req;
+    auto* request = req.add_requests()->mutable_optical_channel_info();
+    request->set_module(module);
+    request->set_network_interface(network_interface);
+    // In-place definition of method retrieving data from generic response
+    // and saving into 'resp' local variable.
+    OpticalChannelInfo resp{};
+    // Writer for retrieving value
+    DataResponseWriter writer([&resp](const DataResponse& in) {
+        if (!in.has_optical_channel_info()) return false;
+        resp = in.optical_channel_info();
+        return true;
+    });
+    // Query the switch. The returned status is ignored as there is no way to
+    // notify the controller that something went wrong. The error is logged when
+    // it is created.
+    // Here we ignore the node_id since it is not valid in this case.
+    tree->GetSwitchInterface()
+        ->RetrieveValue(/*node_id*/ 0, req, &writer, /* details= */ nullptr)
+        .IgnoreError();
+    // Return the retrieved value. Note that we will return a default value if
+    // the second level nest message does not exists.
+    V value;
+    if ((resp.*has_inner_msg_func)()) {
```

Comment:
```
warning: variable 'value' is used uninitialized whenever 'if' condition is false [-Wsometimes-uninitialized]
    if ((resp.*has_inner_msg_func)()) {
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
Let's initialize the value with `V value{};`